src/device-state.ts

```diff
@@ -774,17 +774,15 @@ export class DeviceState extends (EventEmitter as new () => DeviceStateEventEmit
 					retryCount,
 				});
 			} catch (e) {
-				const detailedError = new Error(
-					'Failed to apply state transition steps. ' +
-						e.message +
-						' Steps:' +
-						JSON.stringify(_.map(steps, 'action')),
-				);
-				return this.applyError(detailedError, {
-					force,
-					initial,
-					intermediate,
-				});
+				const detailedError = !(e instanceof UpdatesLockedError)
+					? new Error(
+							'Failed to apply state transition steps. ' +
+								e.message +
+								' Steps:' +
+								JSON.stringify(_.map(steps, 'action')),
+					  )
+					: e;
+				throw detailedError;
```

Comment:
I think it'd be clearer as:
```suggestion
				 if (e instanceof UpdatesLockedError) {
					throw e;
				 }
				throw new Error(
					'Failed to apply state transition steps. ' +
						e.message +
						' Steps:' +
						JSON.stringify(_.map(steps, 'action')),
				);
```